src/utils/internal/isGenerator.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * Determines if the given function is a generator function.
+ */
+export function isGenerator<IteratorType>(
+  fn: any,
+): fn is Generator<IteratorType, IteratorType, IteratorType> {
+  if (!fn) {
+    return false
+  }
+
+  return typeof (fn as Generator<unknown>)[Symbol.iterator] == 'function'
+}
```

Comment:
This might be better named/typed as `isIterable`. The test will return true for anything that has a Symbol.iterator function method; I don't think there's a good way of distinguishing them. I'm also not sure if there's a good reason to disallow iterables that aren't generators, as they should work in the same way without errors.

Reply:
Great suggestion! I've preparing some polishing to the logic and will push it later today.